tempodb/search/backend_search_block.go

```diff
@@ -228,6 +160,27 @@ func (s *BackendSearchBlock) Search(ctx context.Context, p Pipeline, sr *Results
 		return err
 	}
 
+	// Read header
+	// Verify something in the block matches by checking the header
+	hbr, hbrlen, err := s.l.Read(ctx, "search-header", backend.KeyPathForBlock(s.id, s.tenantID), true)
+	if err != nil {
+		return err
+	}
+
+	sr.bytesInspected.Add(uint64(hbrlen))
+
+	hb, err := tempo_io.ReadAllWithEstimate(hbr, hbrlen)
+	if err != nil {
+		return err
+	}
+
+	header := tempofb.GetRootAsSearchBlockHeader(hb, 0)
+	if !p.MatchesBlock(header) {
+		// Block filtered out
+		// TODO - metrics ?
```

Comment:
Yeah I think we should add a metric here to track how effective the block headers are.

Reply:
Some measurement is already possible by comparing the inspectedBytes/Traces/Blocks in the response vs. an exhaustive search with `x-dbg-exhaustive`.  This comment was more about if there is anything missing, like maybe we should count the block header in inspectedBytes so that it more accurately measures i/o performed during the search.

What metric did you have in mind?

Reply:
I was thinking something along the lines of exported metrics (to prometheus) that show how many requests are filtered out by the headers, that would be really useful to track. 

But also thinking if we can make this generic enough and instrument each stage of the pipeline.